frontned/src/components/navbar/DropdownMenu.tsx

```diff
@@ -0,0 +1,121 @@
+import { useState } from 'react';
+import { MouseEvent } from 'react';
+import { styled } from '@mui/material/styles';
+import Button from '@mui/material/Button';
+import Menu, { MenuProps } from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
+import { Divider } from '@mui/material';
+
+const StyledMenu = styled((props: MenuProps) => (
+  <Menu
+    elevation={0}
+    anchorOrigin={{
+      vertical: 'bottom',
+      horizontal: 'right',
+    }}
+    transformOrigin={{
+      vertical: 'top',
+      horizontal: 'right',
+    }}
+    {...props}
+  />
+))(({ theme }) => ({
+  '& .MuiPaper-root': {
+    padding: '0',
+    boxShadow:
+      '0 3px 14px 2px #0000001F, 0 8px 10px 1px #00000024, 0 5px 5px -3px #00000033',
+    '& .MuiMenu-list': {
+      '& .MuiDivider-root': {
+        margin: '0 0',
+      },
+      '& .MuiMenuItem-root': {
+        color: theme.palette.text.primary,
+        width: '220px',
+        padding: '6px 16px',
+      },
+    },
+  },
+}));
+
+const DropdownButton = styled(Button)(({ theme }) => ({
+  backgroundColor: theme.palette.primary.contrastText,
+  color: theme.palette.text.secondary,
+  textTransform: 'none',
+  padding: '6px 8px',
+  '&:hover': {
+    borderRadius: '4px',
+    backgroundColor: '#0000000A',
+  },
+}));
+
+export default function DropdownMenu() {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+
+  const handleClick = (event: MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  return (
+    <>
+      <DropdownButton
+        id="recipe-navbar"
+        aria-haspopup="true"
+        aria-expanded={open ? 'true' : undefined}
+        variant="contained"
+        disableElevation
+        onClick={handleClick}
+        endIcon={<KeyboardArrowDownIcon />}
+      >
+        Recipes
+      </DropdownButton>
+      <StyledMenu
+        id="recipe-navbar-menu"
+        anchorEl={anchorEl}
+        open={open}
+        onClose={handleClose}
+      >
+        <MenuItem onClick={handleClose} disableRipple>
```

Comment:
Extract link to some kind of Array of objects (label, path) and then iterate on array to populate template ;)

Reply:
We want to create navigation using Link component provided by react router
import { Link } from "react-router-dom";

Reply:
I decided to override the default Link from `@mui/materials/Link` with one from `react-router-dom`, also I disabled some default styles for links. For details You can look at `src/theme.ts` and `src/overrides/RouterLink.tsx`.